components/Workout.js

```diff
@@ -1,126 +1,266 @@
-      <Text>Check your <Link href="/profile">profile</Link> to see your progress and to repeat workouts.</Text>
-    </Flex>
-  </div>
+	const { colorScheme } = useMantineColorScheme()
+	const confettiDom = useRef(null)
+	const [active, setActive] = useState(0)
+	const [opened, setOpened] = useState(false)
+	const [sets, setSets] = useState([])
+
+	const handleChange = (value, index) => {
+		const newSets = [...sets.slice(0, index), value, ...sets.slice(index + 1)]
+		setSets(newSets)
+	}
+
+	const changeStep = (update) => {
+		const userWorkout = (user.workouts || []).sort(
+			(a, b) => new Date(b.created_at) - new Date(a.created_at)
+		)[0]
+		const newIndex = active + update
+
+		updateProgress({ index: active, sets })
+		setSets((userWorkout.exercises[newIndex] || {}).sets || [])
+		setActive(newIndex)
+
+		if (active === workout.length - 1 && update > 0) {
+			party.confetti(confettiDom.current, { count: 40 })
+		}
+	}
+
+	return (
+		<>
+			<SharingModal
+				opened={opened}
+				onClose={() => setOpened(false)}
+				numberOfExercise={workout.length}
+			/>
+			<div>
+				<InfoCard>
+					Disclaimer: All exercises and videos are provided by{" "}
+					<a
+						href="https://musclewiki.com/"
+						target="_blank"
+						rel="noopener noreferrer"
+					>
+						musclewiki.com
+					</a>
+				</InfoCard>
+				{active < workout.length && (
+					<Timeline bulletSize={24} lineWidth={2} active={active}>
+						{workout.map((exercise, index) =>
+							index === active ? (
+								<Timeline.Item
+									key={exercise._id}
+									bullet={
+										colorScheme === "dark" && (
+											<div
+												style={{
+													background: "#1A1B1E",
+													width: "20px",
+													height: "20px",
+													borderRadius: " 50%",
+												}}
+											></div>
+										)
+									}
+								>
+									<ActiveExercise
+										exercise={exercise}
+										handleChange={handleChange}
+										sets={sets}
+										changeStep={changeStep}
+										user={user}
+										active={active}
+									/>
+								</Timeline.Item>
+							) : (
+								<Timeline.Item
+									key={exercise._id}
+									bullet={active > index && <IconCheck />}
+								>
+									<Text mr="xs" fw={500}>
+										{exercise.title}
+									</Text>
+								</Timeline.Item>
+							)
+						)}
+					</Timeline>
+				)}
+
+				<Flex
+					direction="column"
+					align="center"
+					style={{ display: active === workout.length ? "flex" : "none" }}
+				>
+					<Title mt="lg" ref={confettiDom}>
+						Workout completed!
+					</Title>
+					<Image
+						mb="md"
+						src="/trophy.png"
+						alt="illustration of a trophy"
+						width={300}
+						height={300}
+					/>
+					<Text>Congrats, you completed your workout! 🎉</Text>
+					<Text>
+						Check your <Link href="/profile">profile</Link> to see your progress
+						and to repeat workouts.
+					</Text>
+					<Button
+						variant="white"
```

Comment:
Let's use variant="outline" here to make it also look good on dark mode :)

Also add mt="md" to add some margin top